New example: Google ADK

KellySit opened the pull request:
## New Example: ADK Agent

Adds a new example demonstrating ADK agent training with Agent-Lightning. ADK is an observability framework for agent traces with span visualization and Cloud Trace integration.

**Files Added:**
- `examples/google_adk/adk_agent.py` - LitAdkAgent implementation
- `examples/google_adk/train_adk.py` - Training script with CI support
- `examples/google_adk/adk_debug.py` - Sanity check script
- `examples/google_adk/prepare_dataset.py` - Dataset conversion utility
- `docs/how-to/train-adk-agent.md` - Comprehensive training guide
- Updated `pyproject.toml` with `adk` optional dependency group

KellySit commented:
@microsoft-github-policy-service agree [company="KellySit"]

ultmaster commented on examples/google_adk/.gitignore:
```diff
@@ -0,0 +1,8 @@
+vendor/
```
I think most of these ignores have been included in global gitignore. Please check,

ultmaster commented on examples/google_adk/adk_debug.py:
```diff
@@ -0,0 +1,70 @@
+console = Console()
+
+
+def parse_args() -> argparse.Namespace:
```
let's work together to add CI for this example.

ultmaster commented on examples/google_adk/README.md:
```diff
@@ -0,0 +1,120 @@
+
+This example demonstrates training a Google ADK (Application Development Kit) agent using Agent-Lightning with reinforcement learning. The agent processes user instructions and generates ADK actions for Google applications. It's compatible with Agent-lightning v0.2 or later.
+
+## Requirements
```
If you have decided to create a how-to tutorial, this README can be greatly simplified and only keeps necessarily installation and running guidance.

ultmaster commented on .gitignore:
```diff
@@ -206,10 +206,4 @@ cython_debug/
-agentlightning/dashboard/**/*.js
-agentlightning/dashboard/**/*.html
-agentlightning/dashboard/**/*.svg
+.claude/*.local.json
```
no need to change this I think.

ultmaster reviewed (commented):
Thank you for your contribution! Let's work together to polish this work!

KellySit commented:
@ultmaster Thank you so much for the feedback. I have made some updates based on the feedback:  
- Fixed unnecessary changes (examples/apo/room_selector_apo.py)
- Updated adk_agent.py and adk_debug.py
- Added training result showcase to docs/how-to/train-adk-agent.md
- Added CI and renewed prepare_dataset.py
- Omitted unnecessary things (docs/how-to/train-adk-agent.md, examples/google_adk/README.md)
 
The changes are now pushed and ready for re-review. The latest commit is the ready one. Thanks.

KellySit commented on docs/how-to/train-adk-agent.md:
```diff
@@ -0,0 +1,395 @@
+# Train ADK Agent with Agent-lightning and VERL
```
Thank you so much! I have updated and made a latest commit based on the feedback. Ready for the review.

KellySit commented on examples/google_adk/adk_agent.py:
```diff
@@ -0,0 +1,215 @@
+        instruction = self._compose_instruction(task)
+        agent_name = task["app_id"] or "adk_agent"
+
+        return LlmAgent(
```
@ultmaster Thanks! I will update this later, adopt the style @agl.rollout and switch to a different dataset. Feel free to edit the PR, improve it, or help with verification, let’s make this project great together!

johnson7788 commented:
Great example, I love you
